src/hooks/wallets/useOnboard.ts

```diff
@@ -74,10 +78,27 @@ export const useInitOnboard = () => {
   useEffect(() => {
     if (!onboard) return
 
-    const walletSubscription = onboard.state.select('wallets').subscribe((wallets) => {
+    const walletSubscription = onboard.state.select('wallets').subscribe(async (wallets) => {
       const newWallet = getConnectedWallet(wallets)
       if (newWallet) {
-        lastWalletStorage.set(newWallet?.label)
+        const { label, provider } = newWallet
+
+        lastWalletStorage.set(label)
+
+        trackEvent({ ...WALLET_EVENTS.CONNECT, label })
+
+        if (label.toUpperCase() === 'WALLETCONNECT') {
+          const { default: WalletConnect } = await import('@walletconnect/client')
+
+          const peerWallet =
+            ((provider as unknown as any).connector as InstanceType<typeof WalletConnect>).peerMeta?.name ||
```

Comment:
Isn't a JsonRpcProvider or something like that.

Reply:
It's [`EthProvider`](https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/walletconnect/src/index.ts#L56). We can't import it sadly. The important part is `WalletConnect` though.